WordPress/gutenberg
# MenuItem: Add 40px size prop on Button

In preparation for #65751

## What?

Adds `__next40pxDefaultSize` prop to the Button in MenuItem.

## Testing Instructions

See the `MenuItem` stories in Storybook. The heights should now be 40px instead of 36px.

<img width="277" alt="MenuItem with 40px height" src="https://github.com/user-attachments/assets/af12448a-4c3a-49a3-8eb5-bacb106d0c7c">

(These changes mostly won't be seen in the app until we update the styles for DropdownMenu, which is currently overriding the MenuItem styles 😅)
